apps/web/src/containers/TradingContest/hooks/use-nft-contract.tsx

```diff
@@ -0,0 +1,45 @@
+import { Contract } from 'ethers';
+import { useState } from 'react';
+import { NftContract, NftContractABI } from '@notional-finance/contracts';
+import { getProviderFromNetwork } from '@notional-finance/util';
+import { useSelectedNetwork } from '@notional-finance/notionable-hooks';
+import { useConnect } from '@notional-finance/wallet/hooks/use-connect';
+
+export enum BETA_ACCESS {
+  PENDING = 'pending',
+  CONFIRMED = 'confirmed',
+  REJECTED = 'rejected',
+}
+
+export const useNftContract = () => {
+  const selectedNetwork = useSelectedNetwork();
+  const { selectedAddress } = useConnect();
+  const [betaAccess, setBetaAccess] = useState<BETA_ACCESS>(
```

Comment:
This is fine for now, but later we will have to gate the entire UI if the beta access is not found. So we will need to put this into global state somehow.